packages/ai-jsx/src/core/completion.tsx

```diff
@@ -142,6 +155,49 @@ export function AssistantMessage({ children }: { children: Node }) {
   return children;
 }
 
+/**
+ * Provide a Function Call to the chat model.
+ *
+ * The function call tells the model a previous function that was invoked by the model. See https://platform.openai.com/docs/guides/gpt/chat-completions-api for more detail.
+ * In case of model returning a function call, ChatCompletion would return a <FunctionCall> component as well.
+ *
+ * This component can only be used within a ChatCompletion.
+ *
+ * @example
+ *    <ChatCompletion>
+ *      <UserMessage>What is 258 * 322?</UserMessage>
+ *      <FunctionCall name="evaluate_math" args={expression: "258 * 322"} />
+ *      <FunctionResponse name="evaluate_math">83076</FunctionResponse>
+ *
+ *    ==> "That would be 83,076."
+ */
+export function FunctionCall({ name, args }: { name: string; args: Record<string, string | number | boolean | null> }) {
```

Comment:
Functions can only be called with `string | number | boolean`? The model can't call a function by passing an object? e.g. `f({a: 123})`

Reply:
Yes the model expects a flat list of _named_ parameters, each of which with a type like that. I didn't see any mention of the full list of supported types in the openai docs. I hypothesize whatever we pass as "type" (which is a string field) will be passed to the model, and the model may or may not do the right thing.

But since the model calls function by named parameters; maybe the javascript equivalent of that will become that gets all arguments as an object `func({param: value})`.